pkg/tarmak/kubectl/kubectl.go

```diff
@@ -190,6 +190,33 @@ func (k *Kubectl) ensureWorkingKubeconfig() (interfaces.Tunnel, error) {
 
 	// check if certificates are set
 	if len(authInfo.ClientCertificateData) == 0 || len(authInfo.ClientKeyData) == 0 || len(cluster.CertificateAuthorityData) == 0 {
+
+		// generate tf code
+		if err := k.tarmak.Terraform().GenerateCode(k.tarmak.Environment().Hub()); err != nil {
+			return nil, err
+		}
+		// symlink tarmak plugins into folder
+		if err := k.tarmak.Terraform().PreparePlugins(k.tarmak.Environment().Hub()); err != nil {
+			return nil, err
+		}
+		// run init
+		if err := k.tarmak.Terraform().Command(
+			k.tarmak.Environment().Hub(),
+			[]string{
+				"terraform",
+				"init",
+				"-get-plugins=false",
+				"-input=false",
+			},
+			nil,
+			nil,
+			nil,
+		); err != nil {
+			return nil, err
+		}
+		// write ssh_config
+		k.tarmak.SSH().WriteConfig()
+
```

Comment:
This is a lot duplicated code from https://github.com/jetstack/tarmak/blob/master/pkg/terraform/terraform.go#L108

This code should be happening in a single method in terraform.go and we just export and call a single method `Prepare()`